README.md

```diff
@@ -888,3 +813,444 @@ Optional.Some(3).tap(x  =>  console.log("Hello") ||  x  +  1);
+Ok(3).reject(x => x === 3)
+
+// This evaluates to Ok(3)
+Ok(3).reject(x => x === 6)
+```
+
+#### `Result`'s `match(callbacks)`
+
+`match` expects an object with one or multiple `Result` types.
+
+```javascript
+// This evaluates to 4
+Ok(3).match({
+  Error:   Result.Error,
+  Aborted: Result.Error,
+  Ok(value) {
+    return value + 1;
+  }
+});
+
+// This evaluates to "Oops"
+Error().match({
+  Error() {
+    return 'Oops';
+  },
+  Ok:      Result.Ok
+});
+
+// This will throw "Oops"
+Error('Oops').match({
+  Aborted: Result.Aborted,
+  Ok:      Result.Ok
+});
+```
+
+#### `Result`'s `recover(λ)`
+
+Unconditionally recovers from an `Error`, does not have any effect on an `Ok`.
+
+```javascript
+// Evaluates to Ok('OK!')
+Error('Oops').recover(() => 'OK!');
+
+// Evaluates to Ok('Yeah!')
+Ok('Yeah!').recover(() => 'OK!');
+```
+
+#### `Result`'s `recoverWhen(predicate, λ)`
+
+`Result`'s `recoverWhen` allows you to recover from an `Error`, if the result of transforming the error through `λ` evaluates to `true`.
+__Attention__: Note that the `λ` function is applied __before__ the `predicate` is evaluated, which may be unexpected.
+
+```javascript
+// This evaluates to Ok(6)
+Error(3).recoverWhen(x => x === 6, x => x * 2);
+
+// Evaluates to Ok(3)
+Ok(3).recoverWhen(x => x === 6, x => x * 2);
+```
+
+#### `Result`'s `abortOnError`
+
+Converts an `Error` into an unrecoverable `Aborted`
+
+```javascript
+// Evaluates to Ok(3)
+Ok(3).abortOnError();
+
+// Evaluates to Aborted('Oops')
+Error('Oops').abortOnError();
+```
+
+#### `Result`'s `abortOnErrorWith(λOrValue)`
+
+Transforms an `Error` into an unrecoverable `Aborted` by substituting or applying `λOrValue`
+
+```javascript
+// Evaluates to Ok(3)
+Ok(3).abortOnErrorWith(4);
+
+// Evaluates to Aborted('You shall not pass!')
+Error('Oops').abortOnErrorWith('You shall not pass!');
+
+// Evaluates to Aborted('Oops, that was a failure')
+Error('Oops').abortOnErrorWith(x => `${x}, that was a failure`);
+```
+
+#### `Result`'s `asynchronous`
+
+__TODO__: Not sure I get this
+
+```javascript
+// This evaluates to createPending // Pending(3) <-- expected?
+Ok(3).asynchronous();
+
+Error('Oops!').asynchronous();
+```
+
+#### `Result`'s `fromPromise`
+
+Any value that represents a `Promise` can be converted into a `Result`. This means you can run any asynchronous functions and wrap them to handle their return value in a functional chain.
+The result will be a `Result` of the `Pending` subtype, eventually resolving to an `Ok` or an `Error` subtype.
+
+```javascript
+const result1 = Result.fromPromise(axios.put(`/api/person/${person}`, person));
+
+const result2 = Result.fromPromise(bcrypt.genSalt(saltRounds));
+```
+
+#### `Result`'s `toPromise`
+
+Just as a promise can be turned into a `Result`, any `Result` can also be turned back into a promise if your caller expects one:
+
+```javascript
+return Result
+.fromPromise(handleUpload(request, pipeFileToStorage))
+.abortOnErrorWith(Response.serverError)
```

Comment:
The response stuff is super confusing for outsiders, maybe use a less fancy arrow function? `() => new Response.serverError()`